lib/spack/spack/environment/__init__.py

```diff
@@ -16,11 +16,12 @@
 The high-level format of a Spack lockfile hasn't changed much between versions, but the
 contents have.  Lockfiles are JSON-formatted and their top-level sections are:
 
-  1. ``_meta`` (object): this contains deatails about the file format, including:
+  1. ``_meta`` (object): this contains details about the file format, including:
       * ``file-type``: always ``"spack-lockfile"``
       * ``lockfile-version``: an integer representing the lockfile format version
       * ``specfile-version``: an integer representing the spec format version (since
         ``v0.17``)
+      * ``spack-commit``: the Spack git commit (since ``v0.19``)
```

Comment:
This doesn't belong in `_meta`.  `_meta` is about the format of the lock file itself (it's really schema details so that we know if this version of spack can read the file).

The `spack-commit` is a new field in the file with "real" data.  It's complicated to talk about this here b/c `spack.lock` is a file full of metadata about the build.  `_meta` is metadata about the file.

From experience: I think we should have a top-level `spack` dictionary to make this more extensible.  Already I know we will want:

```json
"spack": {
    "type": "git",
    "commit": "457daf4be65478a592102177662eb9dad52b8767"
}
```

and:

```json
"spack": {
    "type": "release",
    "version": "0.18.1"
}
```

Reply:
Great!  Thanks for the clarification.

Reply:
As discussed off-line earlier, since it is easy to grab (and can be useful from a human reading the file perspective), will store the version even when storing the commit.